src/PointwiseFunctions/AnalyticSolutions/NewtonianEulerSolutions/IsentropicVortex.cpp

```diff
@@ -0,0 +1,112 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticSolutions/NewtonianEulerSolutions/IsentropicVortex.hpp"
+
+#include <cmath>
+#include <ostream>
+
+#include "DataStructures/DataVector.hpp"  // IWYU pragma: keep
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Parallel/PupStlCpp11.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+/// \cond
+namespace NewtonianEuler {
+namespace Solutions {
+
+IsentropicVortex::IsentropicVortex(
+    const double adiabatic_index, const IsentropicVortex::Center::type center,
+    const IsentropicVortex::MeanVelocity::type mean_velocity,
+    const double strength, const OptionContext& context)
+    : adiabatic_index_(adiabatic_index),
+      center_(center),
+      mean_velocity_(mean_velocity),
+      strength_(strength) {
+  if (adiabatic_index_ <= 1.0 or adiabatic_index_ >= 2.0) {
```

Comment:
Parsing errors are already handled by specifying the bounds of the option, so these can be turned into ASSERTs for when the object is created without specifying options